### ISSUE TITLE
Would be nice to see some date handling logic examples

### ISSUE THREAD

**Issue by gala-dan** [GalateaRaj/fuse-starter-java#60]
Is java.util.Date a sensible type for storing dates, or should we use ZonedDateTime instead? What about dates coming to us in a json format?

Other date related operations?

**Comment by GalateaEric** [comment]
yes, we should be using ZonedDateTime or LocalDateTime from java 8 onwards.  java.util.Date should generally be avoided.

Probably need some way to demonstrate all the usual functions like: https://gist.github.com/salomvary/79977bbdb3a9cccadac2

**Comment by GalateaEric** [comment]
assigning to Danny to come up with a list of all the desired date functionality we should aspire to demonstrate.

**Comment by gala-dan** [comment]
Thoughts:

1. Introduce the concept of a date/calendar service, which would be used to abstract away all date logic anywhere we need it. This makes testing easier as we can ensure determinism. 
2. Converting local dates to UTC
3. General date comparisons
4. Handling of holidays (which may be a separate issue entirely)

**Comment by GalateaEric** [comment]
Alex F would like to add something related to url/spring parameters

**Comment by GalaFirsh** [comment]
Basically what's the best way to go from a url param to a usable java object for dates?  @DateTimeFormat perhaps?

Something where you're adding a time-based filter to a query for example, eg:
``http://host:port/trades?since=2018-03-17``

**Comment by mikegajda** [comment]
1. Store timestamps in UTC
2. Set JVM timezone in UTC
3. ZonedDateTime should be used basically everywhere, and only convert to LocalDateTime at boundary layers, if necessary. In general, return dates with timezone data
4. When sending JSON, send a JSON String, and then parse into a JavaScript Date object


Todo for this ticket: 
- Add the first three items from DateTest.java to the README and point the reader at the test for everything else. Also add the four items above as general practices to follow

Other:
- Dates in Database. We need recommendations on this on how to convert between java.sql.Date into ZonedDateTime. New ticket, out of scope #335 
- How to handle dates on the frontend. Ticket for fuse-starter-typescript, this is now in the fuse-starter-typescript as ticket 12 https://github.com/galatea-associates/fuse-starter-typescript/issues/12
